src/sentry/sentry_metrics/indexer/postgres_v2.py

```diff
@@ -144,11 +144,19 @@ def bulk_record(
             filtered_db_write_keys = writes_limiter_state.accepted_keys
             del db_write_keys
 
+            rate_limited_write_results = writes_limiter_state.dropped_strings
+
+            db_write_key_results = KeyResults()
+            for dropped_string in rate_limited_write_results:
+                db_write_key_results.add_key_result(
+                    dropped_string.key_result,
+                    fetch_type=dropped_string.fetch_type,
+                    fetch_type_ext=dropped_string.fetch_type_ext,
+                )
+
             if filtered_db_write_keys.size == 0:
                 indexer_cache.set_many(new_results_to_cache, use_case_id.value)
-                return cache_key_results.merge(db_read_key_results)
-
-            rate_limited_write_results = writes_limiter_state.dropped_strings
+                return cache_key_results.merge(db_read_key_results).merge(db_write_key_results)
```

Comment:
so does this mean `db_write_key_results` can be a combination of dropped write keys and then also successful results? Would it be weird to have another `KeyResults` that is `db_dropped_write_keys`? Maybe that's too many merges at the bottom 

```python
return cache_key_results.merge(db_read_key_results).merge(db_write_key_results).merge(db_dropped_key_results)
```

Reply:
that makes sense, fixed. curious how this will affect performance...

Reply:
yeah wasn't sure about the overhead of creating a new `KeyResults` would be... maybe we will find out